adapters/v1/syft.go

```diff
@@ -0,0 +1,95 @@
+package v1
+
+import (
+	"bytes"
+	"context"
+	"sync"
+
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/formats/spdxjson"
+	"github.com/anchore/syft/syft/pkg/cataloger"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+	"github.com/kubescape/kubevuln/core/domain"
+	"github.com/kubescape/kubevuln/core/ports"
+	"go.opentelemetry.io/otel"
+)
+
+type SyftAdapter struct {
+	mu sync.Mutex
+}
+
+var _ ports.SBOMCreator = (*SyftAdapter)(nil)
+
+func NewSyftAdapter() *SyftAdapter {
+	return &SyftAdapter{}
+}
+
+func (s *SyftAdapter) CreateSBOM(ctx context.Context, imageID string, options domain.RegistryOptions) (domain.SBOM, error) {
+	ctx, span := otel.Tracer("").Start(ctx, "CreateSBOM")
+	defer span.End()
+	// ensure only one SBOM is created at a time
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	// translate business models in syft models
+	userInput := "registry:" + imageID
+	sourceInput, err := source.ParseInput(userInput, "", true)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
question: is there a reason we panic here, instead of returning the `err`?

Is this a truly unrecoverable error that the whole application has to drop dead? If I am trying to wrap SBOM generation in an HTTP API and a user provides me a malformed input, I would expect to return them an error that tells them their input is invalid, not terminate on the spot.